### PR TITLE
fix bug with google button rendering #131

### LINKED ISSUE #131: [Sign in/Sign up] The size of the “Sign in/Sign up with Google”
Description:
The button is incorrectly sized and redrawn when a modal window is opened

Reproducible: always.

Preconditions:
The user is not logged in.

Steps to reproduce:
open a modal window

Image

Expected result:
The size of the “Sign up with Google” button and the “or continue” description is displayed aligned on the “Login” pop-up.

### PR DESCRIPTION
**Pull request by Mygal-Alex**
https://github.com/Space2Study-Project/Space2Study-FrontEnd/assets/101183408/001c7a69-fd47-410b-9593-05f7dd68ce5c

### CODE REVIEW

**Review by ArtemHolikov — APPROVED** [review]
👍

**Review by AndriiNasadiuk — APPROVED** [review]
👌

**Review by marianamartsy — COMMENTED** [review]
👍

File: src/containers/guest-home-page/signup-dialog/SignupDialog.jsx [hunk 1]
```diff
@@ -85,7 +86,14 @@ const SignupDialog = ({ type }) => {
...
   }, [isDirty, setNeedConfirmation])
-
+  const { isSmallScreen, isMediumScreen, isLargeScreen, isXLargeScreen } =
+    useBreakpointsGoogle()
+  const getButtonWidth = () => {
+    if (isSmallScreen) return 330
+    if (isMediumScreen) return 333
+    if (isLargeScreen) return 340
+    if (isXLargeScreen) return 394
+  }
```
Comment (vlad-khrychov) [inline src/containers/guest-home-page/signup-dialog/SignupDialog.jsx:96]: We can use `useCallback` or object style for definition of size like
```suggestion
  const getButtonWidth =useMemo(()=> ({
    isSmallScreen: 330,
    isMediumScreen: 333,
    isLargeScreen: 340,
    isXLargeScreen: 394
  }), [] )
```

File: src/containers/guest-home-page/signup-dialog/SignupDialog.jsx [hunk 2]
```diff
@@ -111,7 +119,7 @@ const SignupDialog = ({ type }) => {
             handleSubmit={handleSubmit}
           />
           <GoogleLogin
-            buttonWidth={styles.form.maxWidth}
+            buttonWidth={getButtonWidth()}
```
Comment (vlad-khrychov) [inline src/containers/guest-home-page/signup-dialog/SignupDialog.jsx:122]: If you agree with object style: 
```suggestion
            buttonWidth={getButtonWidth[sizeName]}
```

File: src/containers/guest-home-page/signup-dialog/SignupDialog.jsx [hunk 3]
```diff
@@ -85,7 +86,14 @@ const SignupDialog = ({ type }) => {
   useEffect(() => {
     setNeedConfirmation(isDirty)
   }, [isDirty, setNeedConfirmation])
-
+  const { isSmallScreen, isMediumScreen, isLargeScreen, isXLargeScreen } =
```
Comment (vlad-khrychov) [inline src/containers/guest-home-page/signup-dialog/SignupDialog.jsx:89]: Here we have only one `true` why we need to obtain 4 variables? Mayme we can obtain only one `sizeName`?

**Review by vlad-khrychov — APPROVED** [review]
Think about suggestions. Add `useCallback` for now and after that you can merge.

**Review by LeonidShamarin — APPROVED** [review]

**Pull request merged**